File: sorting.java
class bubble_sort{//time complexity O(n^2)
    public static int[] bubble(int[] arr){
        for(int i=0;i<arr.length-1;i++){
            for(int j=0;j<arr.length-i-1;j++){
                if(arr[j]>arr[j+1]){
                    int t=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=t;
                }
            }
        }
        return arr;
    }
}

class selection_sort{//time complexity O(n^2)
    public static int[] selection(int[] arr){
        for(int i=0;i<arr.length-1;i++){
            int smallest=i;
            for(int j=i+1;j<arr.length;j++){
                if(arr[smallest]>arr[j]){
                    smallest=j;
                }
            }
            int t=arr[smallest];
            arr[smallest]=arr[i];
            arr[i]=t;

        }
        return arr;
    }
}

class merge_sort{//time complexityo(nlogn)
    public static void conquer(int si,int mid,int ei,int arr[]){
        int m_array[]= new int[ei-si+1];
        int l=si;
        int r=mid+1;
        int k=0;
        while(l<=mid && r<=ei){
            if (arr[l]<arr[r]){
                m_array[k]=arr[l];
                l++;
                k++;
            }
            else{
                m_array[k]=arr[r];
                r++;
                k++;
            }
        }

        while(l<=mid){
            m_array[k]=arr[l];
            l++;
            k++;
        }
        while(r<=ei){
            m_array[k]=arr[r];
            r++;
            k++;
        }
        for(int i=0,j=si;i<m_array.length-1;j++,i++){
            arr[j]=m_array[i];
        }
    }
    public static void divide(int si,int ei,int arr[]){
        if(si>=ei){
            return;
        }
        int mid=(si+(ei-si)/2);
        divide(si,mid,arr);
        divide(mid+1,ei,arr);
        conquer(si, mid, ei, arr);
    }
    public static int[] merge(int arr[]){
        int n=arr.length;
        merge_sort.divide(0,n-1,arr);
        return arr;
    }

}
class sorting{
    public static void printarray(int arr[]){
        for(int i=0;i<arr.length;i++){
            System.out.println(arr[i]+" ");
        }
    }
    public static void main(String[] args) {
        int[] array={85,22,64,32,78,50};
        int[] b_array=bubble_sort.bubble(array);
        System.out.println("Bubble Sorted");
        printarray(b_array);
        int[] s_array=selection_sort.selection(array);
        System.out.println("selection Sorted");
        printarray(s_array);
        int[] m_array=merge_sort.merge(array);
        System.out.println("Merge Sorted");
        printarray(m_array);
    }
}

File: sorting.py

def bubble(arr):
    for i in range (0,len(arr)-1):
        for j in range (0,len(arr)-1-i):
            if arr[j]>arr[j+1]:
                t=arr[j]
                arr[j]=arr[j+1]
                arr[j+1]=t
            
    return arr

    
def selection(arr):
    for i in range (0,len(arr)-1):
        smallest=i
        for j in range(i,len(arr)-1):
            if (arr[smallest]>arr[j]):
                smallest=j
        t=arr[i]
        arr[i]=arr[smallest]
        arr[smallest]=t
    return arr
def conquer(si,ei,mid,arr):
    
    n1=mid-si+1
    n2=ei-mid
    i=j=0
    k=si
    left=arr[:n1]
    right=arr[mid+1:]

    while(i<=n1 and j<=n2):
        if (left[i]<right[j]):
            arr[k]=left[i]
            i+=1
            k+=1
            
        else:
            arr[k]=right[j]
            j+=1
            k+=1
                   

    while(i<=n1):
        arr[k]=left[i]
        i+=1
        k+=1
        
    while(j<=n2):
        arr[k]=right[j]
        j+=1
        k+=1
     
        
    
def divide(si,ei,arr):
    if(si>=ei):
        return 
    mid=si+(ei-si)/2
    divide(si,mid,arr)
    divide(mid+1,ei,arr)
    conquer(si,ei,mid,arr)

def merge(arr):
    divide(0,5,arr)

array=[54,23,67,12,98,32]
b_array=bubble(array)
print(b_array)
s_array=selection(array)
print(s_array)

merge(array)
print(array)